rosbag2_tests/test/rosbag2_tests/record_fixture.hpp

```diff
@@ -140,6 +141,7 @@ class RecordFixture : public TemporaryDirectoryFixture
 
   std::string bag_path_;
   std::string database_path_;
+  std::string storage_path_;
```

Comment:
What's the difference between these three variables now? What's the difference between `database_path_` and `storage_path_`?

Reply:
`Writer` consumes `bag_path_`, `SqliteStorage` consumes `storage_path_`, and `SqliteWrapper` can consume `database_path_`

This is related to what I commented here: https://github.com/ros2/rosbag2/pull/185#issuecomment-543895687.

The e2e test in `rosbag2_tests` opens the database either through `Writer`, `SqliteStorage`, or `SqliteWrapper`. 
`SqliteWrapper` hasn't changed and expects the path to the db, which is `database_path_`
`SqliteStorage` now also expects the db as well, but without the extension since that may be plugin-specific. This is `storage_path_`
`Writer` expects the base path, which is `bag_path_`.

Reply:
> SqliteStorage now also expects the db as well, but without the extension since that may be plugin-specific. This is storage_path_

I don't think I understand this. In the metadata.yaml, there should be a path to an existing file listed in `relative_file_paths`. This filepath should then be passed to a `Reader` which opens it through the appropriate plugin. The Sqlite storage plugin should thus be able to handle a file which ends in `.db3`.